### DISCUSSION TITLE
correct type for SerialWrite?

### DISCUSSION THREAD

**Discussion by nghess** [bonsai-rx/bonsai#1612] [Q&A] [answered]
Hi,
I'm having trouble sending a SoftCode to Bpod using the SerialWrite sink. I can't seem to coerce an int (or any other type) into a byte format that SerialWrite agrees with. Can anyone provide me with some guidance?

Thanks for any help!

**Comment by bruno-f-cruz** [comment, marked as answer]
If you want to cast it, you can use an `InputMapping` that maps the value onto the property of the byte:
![image](https://github.com/bonsai-rx/bonsai/assets/7049351/bbaf66a0-bd48-42f0-9c93-5fbb9b038451)

You can also just use an `ExpressionTransform` and leverage the [`Convert` class](https://learn.microsoft.com/en-us/dotnet/api/system.convert.tobyte?view=net-7.0)

![image](https://github.com/bonsai-rx/bonsai/assets/7049351/f170878c-4c35-4dc1-8746-1904f96e8619)

If you want the full raw byte[] representation of your data type, then I would probably write my own c# script using the [BitConverter](https://learn.microsoft.com/en-us/dotnet/api/system.bitconverter?view=net-7.0) class.

Hope it helps

**Reply by nghess** [reply]
Thanks so much Bruno! Extremely helpful answer.

**Reply by nghess** [reply]
Update: The SerialWrite example does work, however it only seems to send the byte once, freezing it at its initial value, whereas my graph should be updating at 80hz (camera fps). 

Tinkering around, I was able to get the byte to update as expected if using a TakeLast, however the SerialWrite no longer seems to be sending. Any tips?

Thanks again for the help!

**Reply by bruno-f-cruz** [reply]
Toarray operates when the sequence ends. If you want to send multiply bytes you will need to move the conversion inside a SelectMany.

**Reply by nghess** [reply]
Worked! Thanks again.
